conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Parameters.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.collect.Multimap;
+import com.palantir.logsafe.SafeArg;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.AttachmentKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class to put different types of safe or unsafe parameters in a MultiMap attached to the exchange.
+ * It requires the exchange to have been priorly through a ParametersHandler which would have created the attached
+ * MultiMaps.
+ */
+public final class Parameters {
+
+    private static final Logger log = LoggerFactory.getLogger(Parameters.class);
+
+    public static final AttachmentKey<Multimap<String, Object>> SAFE_PARAMS_ATTACH_KEY =
+            AttachmentKey.create(Multimap.class);
+    public static final AttachmentKey<Multimap<String, Object>> UNSAFE_PARAMS_ATTACH_KEY =
```

Comment:
Is there any value in logging unsafe components? Why not only add safe ones?

Reply:
WC was logging unsafe query and path params. I think the biggest value is CI and on-prem debugging.

Reply:
If I'm reading this correctly, we're logging everything as unsafe, where I think wc only logs things annotated with `@Unsafe` as unsafe parameters, right? We shouldn't log credential data as unsafe by default, because even logging a BearerToken (or other non-bearer credential) to disk is dangerous.

Reply:
We are not logging AuthHeader in WC but WC consider every query and path params not safe as unsafe and log them. It is not taking into account any `@Unsafe` notation.

Reply:
Let me actually make that more precise, we do log auth params but only those in a whitelist.

Reply:
Our header logging isn't connected to service parameter logging though. Our whitelisted set of headers are already logged regardless of endpoint.

Are you sure that path parameters area default logged as unsafe? I had thought they were omitted, but I may be misremembering.

Reply:
re: path parameters default:

That test snippet in our WC tests is relevant
```
        assertThat(requestLog2.getUnsafeParams()).contains(
                Pair.of("path2", "path2V"),
                Pair.of("query", "query1V"));
```